gtsam/discrete/DecisionTreeFactor.cpp

```diff
@@ -536,5 +536,11 @@ namespace gtsam {
     return DecisionTreeFactor(this->discreteKeys(), thresholded);
   }
 
+/* ************************************************************************ */
+DiscreteFactor::shared_ptr DecisionTreeFactor::restrict(
+    const DiscreteValues& assignment) const {
+  throw std::runtime_error("DecisionTreeFactor::restrict not implemented");
```

Comment:
This should just be `Base::restrict` and convert to a factor, right?

Reply:
Hmm, is already a factor, so it should restrict using the fact that it is a decision tree, and then return a restricted decision tree. Base::restrict can’t do that.